### PR TITLE
Implement navigation to profiles and chats from contact requests inside Activity Center

### LINKED ISSUE #19649: Implement navigation to profiles and chats from contact requests inside Activity Center
### Summary

* When someone opens the activity center, and views their contact requests, then pressing on the contact tag should open a profile and/or a 1-1 chat?
  * At the moment:
    * an accepted contact request will navigate to 1-1 chat when the contact tag is pressed
    * a pending contact request will not navigate anywhere when the contact tag is pressed
    * a declined contact request will not navigate anywhere when the contact tag is pressed
    * a contact acceptance notification will not navigate anywhere when the contact is pressed
  * Ideally:
    * an accepted contact request will navigate to 1-1 chat when the contact tag is pressed
    * a pending contact request will navigate to the contact profile contact tag is pressed
    * a declined contact request will navigate to the contact profile when the contact tag is pressed
    * a contact acceptance notification will navigate to the contact profile when the contact is pressed

### Example

https://github.com/status-im/status-mobile/assets/2845768/416cfe94-e494-4a5e-bdf3-cc8057bfd553

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/19649

### Testing Note:
Please test navigation behavior as per requirements mentioned in https://github.com/status-im/status-mobile/issues/19649

status: ready

### CODE REVIEW

**Comment by Parveshdhull** [comment]
@seanstrom Thank you for creating issue https://github.com/status-im/status-mobile/issues/19649, I have implemented navigation as per requirements mentioned in issue. I have two questions

1. Issue doesn't mention what should happen to activity center when we navigate to contact profile. So In this PR I am just opening profile screen over activity screen. Please check out PR and let me know if this is expected behavior or activity center should be closed?
2. `an accepted contact request will navigate to 1-1 chat when the contact tag is pressed`. All other contact tags are navigating to user profile, this will look little inconsistent. Should we navigate to chat only when whole activity center log is pressed and in this case also simply open profile when contact tag is pressed? We can simply navigate to chat from opened profile screen. wdyt?

**Comment by Parveshdhull** [comment]
@status-im/mobile-qa Please let me know if you have any opinion on [this](https://github.com/status-im/status-mobile/pull/19902#issuecomment-2094709013)

File: src/status_im/contexts/shell/activity_center/notification/common/view.cljs [hunk 1]
```diff
@@ -12,11 +12,14 @@
 (defn user-avatar-tag
   [user-id]
   (let [profile (rf/sub [:contacts/contact-by-identity user-id])]
-    [quo/context-tag
-     {:blur?           true
-      :size            24
-      :full-name       (profile.utils/displayed-name profile)
-      :profile-picture (profile.utils/photo profile)}]))
+    [rn/view
+     {:on-start-should-set-responder
```
Comment (flexsurfer) [inline src/status_im/contexts/shell/activity_center/notification/common/view.cljs:16]: could you pls elaboeate on how on-start-should-set-responder works here? thanks

Comment (Parveshdhull) [reply src/status_im/contexts/shell/activity_center/notification/common/view.cljs:16]: hi @flexsurfer, Thank you very much for reviewing PR.

We have the whole activity log as pressable for accepted requests, but when context-tag is pressed we only want that to call an on-press event and avoid propagation to parent pressable.

I picked this solution from the [stackoverflow answer](https://stackoverflow.com/questions/31866671/how-to-stop-touch-event-propagation-in-react-native), apparently adding `onStartShouldSetResponder` and returning true is enough for our situation to make it work.

**Comment by seanstrom** [comment]
> @seanstrom Thank you for creating issue #19649, I have implemented navigation as per requirements mentioned in issue. I have two questions
> 
> 1. Issue doesn't mention what should happen to activity center when we navigate to contact profile. So In this PR I am just opening profile screen over activity screen. Please check out PR and let me know if this is expected behavior or activity center should be closed?
> 2. `an accepted contact request will navigate to 1-1 chat when the contact tag is pressed`. All other contact tags are navigating to user profile, this will look little inconsistent. Should we navigate to chat only when whole activity center log is pressed and in this case also simply open profile when contact tag is pressed? We can simply navigate to chat from opened profile screen. wdyt?

Thanks for the questions 🙏

1. I think the activity-center would be closed and then we navigate to the next screen, and I think that would match what we currently have when navigating to the 1-1 chat from the activity-center.
2. Yeah good points, it would feel more consistent to always open the contact profile after pressing the contact tag. I think we should ask the design team if it's okay to make it always open the contact profile 👍 

Side note:
* We could have a press handler on the activity-center log entry, but I wonder if that would lead to some accidental presses in the wrong area. thoughts?

**Comment by Parveshdhull** [comment]
Thank you @seanstrom for quick response
1. Updated PR so that we close activity center when navigating to profile screen.
2. Pinged design team for their input https://discord.com/channels/624634427930312714/928625369542713396/1236991903493525524

> We could have a press handler on the activity-center log entry, but I wonder if that would lead to some accidental presses in the wrong area. thoughts?

Please can you elaborate. We do have a on-press handler for accepted contact requests that navigates to chat screen. Do you mean we add this on-press entry for all activity logs?

**Comment by Parveshdhull** [comment]
> I think we should ask the design team if it's okay to make it always open the contact profile

[confirmed](https://discord.com/channels/624634427930312714/928625369542713396/1236995573979283607) with the design team, Lets always navigate to contact profile for consistency.

**Comment by Parveshdhull** [comment]
@flexsurfer @seanstrom Thank you for reviewing PR, Please let me know if anything else needs to be addressed.

**Review by seanstrom — APPROVED** [review]
Great stuff, thank you for doing this 🙌
Just tested it on my side too with no issues ✅

**Comment by VolodLytvynenko** [comment]
hi @Parveshdhull could you please resolve existing conflicts? Thanx

**Comment by VolodLytvynenko** [comment]
Hi @Parveshdhull, I have a question. Should the 'contact icon' be interactive?
image

Currently, tapping the 'contact icon' for accepted contacts navigates to a 1-1 chat, while it does not navigate anywhere if tapped for pending or declined contact requests.
I would expect it to always navigate to the profile, similar to how it works with the 'contact' tag. WDYT?

**Comment by Parveshdhull** [comment]
Thank you very much @VolodLytvynenko for picking the PR for testing

> hi [@Parveshdhull](https://github.com/Parveshdhull) could you please resolve existing conflicts?

done

> I would expect it to always navigate to the profile, similar to how it works with the 'contact' tag. WDYT?

I am not sure. We probably should discuss this with the design team. But to me, these icons don't look like pressable buttons. They just look like simple indicators to highlight notification types (contact request-related notifications).

> Currently, tapping the 'contact icon' for accepted contacts navigates to a 1-1 chat

When we have accepted contact requests we are making the whole activity log pressable (which includes the contact icon). I am not sure why this was implemented this way and if this is what we wanted.

Sorry probably the issue title is a little misleading, but I think for the scope of this issue/PR we can only focus on navigation from the activity center using contact tags. For other components, we currently don't have guidelines and expected behavior.

**Comment by VolodLytvynenko** [comment]
Hi @Parveshdhull thank you for PR and for clarification. No issues from my side. PR is ready to be merged

**Pull request merged**